File: Core/value/doublevalue.cpp
#include "doublevalue.h"

DoubleValue::DoubleValue() : ValueBase() {
}

DoubleValue::DoubleValue(ValueGroup *valueGroup, QString id, VALUE_TYPE valueType, bool alwaysEmit, QObject *parent) : ValueBase(valueGroup, id, valueType, alwaysEmit, parent)
{

}

QVariant DoubleValue::_updateValue(QVariant newValue) {
    if (newValue.canConvert(QVariant::Double)) {
        return QVariant::fromValue(newValue);
    } else {
        return QVariant();
    }
}

File: AndroidApp/OshCore/src/main/java/com/osh/value/DoubleValue.java
package com.osh.value;

import com.osh.SerializationSupport;

import org.apache.commons.lang3.math.NumberUtils;

public class DoubleValue extends ValueBase<DoubleValue, Double> implements SerializationSupport {

	@Override
	protected Double _updateValue(Object newValue) {
		if (newValue instanceof Double) {
			return (Double) newValue;
		} else if (newValue instanceof Number) {
			return ((Number) newValue).doubleValue();
		} else if (newValue instanceof String) {
			return Double.parseDouble((String) newValue);
		} else {
			return null;
		}
	}

	public DoubleValue(ValueGroup valueGroup, String id, ValueType valueType) {
		super(valueGroup, id, valueType);
	}

	public DoubleValue() {
	}

}
